sklearn/utils/estimator_checks.py

```diff
@@ -355,10 +358,29 @@ def _is_32bit():
     return struct.calcsize('P') * 8 == 32
 
 
+def gram_matrix_if_pairwise(X, estimator, kernel=linear_kernel):
```

Comment:
it would be nice to have a less specific name for this... maybe just `maybe_pairwise(X, estimator)`